userdetails-cognito/src/main/groovy/au/org/ala/userdetails/CognitoPasswordOperations.groovy

```diff
@@ -2,24 +2,25 @@ package au.org.ala.userdetails
 
 import au.org.ala.auth.PasswordResetFailedException
 import au.org.ala.users.UserRecord
+import au.org.ala.web.OidcClientProperties
 import com.amazonaws.services.cognitoidp.AWSCognitoIdentityProvider
 import com.amazonaws.services.cognitoidp.model.AdminInitiateAuthRequest
 import com.amazonaws.services.cognitoidp.model.AdminResetUserPasswordRequest
 import com.amazonaws.services.cognitoidp.model.AdminSetUserPasswordRequest
 import com.amazonaws.services.cognitoidp.model.AuthFlowType
 import com.amazonaws.services.cognitoidp.model.ConfirmForgotPasswordRequest
-import grails.core.GrailsApplication
 import groovy.util.logging.Slf4j
 import org.apache.commons.codec.digest.HmacAlgorithms
 import org.apache.commons.codec.digest.HmacUtils
+import org.springframework.beans.factory.annotation.Autowired
 
 @Slf4j
 class CognitoPasswordOperations implements IPasswordOperations {
 
-    GrailsApplication grailsApplication
-
     AWSCognitoIdentityProvider cognitoIdp
     String poolId
+    @Autowired
```

Comment:
I would prefer no `@Autowired` in here, just make it a constructor parameter.

Reply:
Fixed.